### PR TITLE
Fix panic on Shutdown when there is at least one active connection

### PR DESCRIPTION
**Pull request by hotafrika**
This is the fix for the "panic on sse-server shutdown when there is at least one active connection" issue.
It does the following: 1) after calling the Shutdown() function it doesn't allow new sse-connections to be established 2) it avoids sending to already closed **removeClient** channel

### CODE REVIEW

File: sse.go [hunk 1]
```diff
@@ -128,7 +138,7 @@ func (s *Server) Restart() {
 
 // Shutdown performs a graceful server shutdown.
 func (s *Server) Shutdown() {
-	s.shutdown <- true
+	close(s.shutdown)
```
Comment (insano10) [inline sse.go:141]: Why does this need to change? If no message is sent to the shutdown channel, how will the listeners know the shutdown has occurred? (e.g. ln.80)

File: sse.go [hunk 2]
```diff
@@ -255,7 +265,6 @@ func (s *Server) dispatch() {
 			close(s.addClient)
 			close(s.removeClient)
 			close(s.closeChannel)
-			close(s.shutdown)
```
Comment (insano10) [inline sse.go:258]: Why did this move to shutdown function? The existing behaviour made sense to me.
1. Shutdown function is called which sends an event to the shudown channel
2. dispatch function receives the shutdown event and then closes the channel as no further events are expected

File: sse.go (hunk 1 above)
Comment (hotafrika) [reply sse.go:141]: Point №1: Actually, we never need the value sent to the `shutdown` channel. Actually, we can change the type of this channel from `chan bool` to `chan struct{}`.
Point №2: When we close the channel, then all the readers from this channel (<-s.shutdown) receive the default value (in this case `false` value).

Conclusion: closing the `shutdown` channel will work as a mechanism of informing all the goroutines about the sse server shutdown process.

File: sse.go (hunk 2 above)
Comment (hotafrika) [reply sse.go:258]: As you can see, in this PR all the serving goroutines (ServerHTTP) use reading from the s.shutdown channel. At the same time,  the`dispatch` function also reads from this channel. It means, that if we just send some value to this channel, we don't know where this value will be read: in the `dispatch` function or in the one of running `ServeHTTP` functions. If we close this channel, then all the functions will receive default values by reading from this channel.

Comment (hotafrika) [reply sse.go:258]: @insano10 Please, take a look at another PR (#2) where I added a unit test for the entire shutdown flow. This test fails with the current implementation and succeeds with the updated implementation.

File: sse.go (hunk 1 above)
Comment (insano10) [reply sse.go:141]: Ah oh, I am not experienced in using channels and I didn't know the default value got sent when a channel is closed.

File: sse.go (hunk 2 above)
Comment (insano10) [reply sse.go:258]: Thank you for the explanation, that makes sense now :)

**Review by insano10 — APPROVED** [review]
LGTM

**Pull request merged**